src/monitor/node_active_protocol.c

```diff
@@ -329,30 +342,17 @@ node_active(PG_FUNCTION_ARGS)
  * NodeActive reports the current state of a node and returns the assigned state.
  */
 static AutoFailoverNodeState *
-NodeActive(char *formationId, char *nodeHost, int32 nodePort,
-		   AutoFailoverNodeState *currentNodeState)
+NodeActive(char *formationId, AutoFailoverNodeState *currentNodeState)
 {
 	AutoFailoverNode *pgAutoFailoverNode = NULL;
 	AutoFailoverNodeState *assignedNodeState = NULL;
 
-	pgAutoFailoverNode = GetAutoFailoverNode(nodeHost, nodePort);
+	pgAutoFailoverNode = GetAutoFailoverNodeById(currentNodeState->nodeId);
+
 	if (pgAutoFailoverNode == NULL)
 	{
-		ereport(ERROR, (errmsg("node %s:%d is not registered",
-							   nodeHost, nodePort)));
-	}
-	else if (strcmp(pgAutoFailoverNode->formationId, formationId) != 0)
-	{
-		ereport(ERROR, (errmsg("node %s:%d does not belong to formation %s",
-							   nodeHost, nodePort, formationId)));
-	}
```

Comment:
This check looks like it should still be done

Reply:
Good point, re-adding it now!

Reply:
Done.